apollo-router/src/graphql.rs

```diff
@@ -160,6 +161,106 @@ impl Error {
     }
 }
 
+#[cfg(feature = "apollo_unsupported")]
+#[buildstructor::buildstructor]
+impl Error {
+    /// Returns a builder that builds a GraphQL [`Error`] from its components.
+    ///
+    /// Builder methods:
+    ///
+    /// * `.message(impl Into<`[`String`]`>)`
+    ///   Required.
+    ///   Sets [`Error::message`].
+    ///
+    /// * `.locations(impl Into<`[`Vec`]`<`[`Location`]`>>)`
+    ///   Optional.
+    ///   Sets the entire `Vec` of [`Error::locations`], which defaults to the empty.
+    ///
+    /// * `.location(impl Into<`[`Location`]`>)`
+    ///   Optional, may be called multiple times.
+    ///   Adds one item at the end of [`Error::locations`].
+    ///
+    /// * `.path(impl Into<`[`Path`]`>)`
+    ///   Optional.
+    ///   Sets [`Error::path`].
+    ///
+    /// * `.extensions(impl Into<`[`serde_json_bytes::Map`]`<`[`ByteString`]`, `[`Value`]`>>)`
+    ///   Optional.
+    ///   Sets the entire [`Error::extensions`] map, which defaults to empty.
+    ///
+    /// * `.extension(impl Into<`[`ByteString`]`>, impl Into<`[`Value`]`>)`
+    ///   Optional, may be called multiple times.
+    ///   Adds one item to the [`Error::extensions`] map.
+    ///
+    /// * `.build()`
+    ///   Finishes the builder and returns a GraphQL [`Error`].
+    #[builder(visibility = "pub")]
+    fn new(
+        message: String,
+        locations: Vec<Location>,
+        path: Option<Path>,
+        extension_code: Option<String>,
+        // Skip the `Object` type alias in order to use buildstructor’s map special-casing
+        mut extensions: JsonMap<ByteString, Value>,
+    ) -> Self {
+        if let Some(code) = extension_code {
+            extensions.entry("code").or_insert_with(|| code.into());
+        }
+
+        Self {
+            message,
+            locations,
+            path,
+            extensions,
+        }
+    }
+
+    pub(crate) fn from_value(service_name: &str, value: Value) -> Result<Error, FetchError> {
```

Comment:
@o0Ignition0o forgot to mention that this function would be very useful to have `pub` if possible. It's super useful to handle responses for subgraphs in a custom client plugin.

Reply:
I don't expect this function to change anytime soon so I'll make it public right now, and discuss it with the team early next week.